velox/functions/prestosql/benchmarks/URLBenchmark.cpp

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+#include "folly/Uri.h"
+#include "velox/expression/tests/VectorFuzzer.h"
+#include "velox/functions/Macros.h"
+#include "velox/functions/Registerer.h"
+#include "velox/functions/lib/benchmarks/FunctionBenchmarkBase.h"
+#include "velox/functions/prestosql/SimpleFunctions.h"
+
+namespace {
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::functions;
+
+template <typename Fn, typename TOutString, typename TInputString>
```

Comment:
Do you need TOutString and TInputString? Can you use StringView instead?

Reply:
I think we are supposed to use high level SQL types(Varchar/Varbinary) and cant use the translated types directly. I tried doing that but I get a resolver error during registration.